File: app/src/main/java/org/torproject/android/ui/NoPersonalizedLearningEditText.java
package org.torproject.android.ui;

import android.content.Context;
import androidx.appcompat.widget.AppCompatEditText;
import android.util.AttributeSet;
import android.view.inputmethod.EditorInfo;

public class NoPersonalizedLearningEditText extends AppCompatEditText {
    public NoPersonalizedLearningEditText(Context context, AttributeSet attrs) {
        super(context, attrs);
        setImeOptions(getImeOptions() | EditorInfo.IME_FLAG_NO_PERSONALIZED_LEARNING);
    }
}

File: appcore/src/main/java/org/torproject/android/core/ui/NoPersonalizedLearningEditText.kt
package org.torproject.android.core.ui

import android.content.Context
import android.os.Build
import android.util.AttributeSet
import android.view.inputmethod.EditorInfo
import androidx.annotation.RequiresApi
import androidx.appcompat.widget.AppCompatEditText

@RequiresApi(Build.VERSION_CODES.O)
class NoPersonalizedLearningEditText(context: Context, attrs: AttributeSet?) : AppCompatEditText(context, attrs) {
    init {
        imeOptions = imeOptions or EditorInfo.IME_FLAG_NO_PERSONALIZED_LEARNING
    }
}